module/x/gravity/abci.go

```diff
@@ -100,12 +102,30 @@ func pruneSignerSetTxs(ctx sdk.Context, k keeper.Keeper) {
 		earliestToPrune := currentBlock - params.SignedSignerSetTxsWindow
 		for _, set := range k.GetSignerSetTxs(ctx) {
 			if set.Nonce < lastObserved.Nonce && set.Height < earliestToPrune {
+				k.DeleteEthereumSignatures(ctx, set.GetStoreIndex())
 				k.DeleteOutgoingTx(ctx, set.GetStoreIndex())
 			}
 		}
 	}
 }
 
+// pruneEventVoteRecords deletes all event vote records with nonces that are older than the last observed event nonce
+func pruneEventVoteRecords(ctx sdk.Context, k keeper.Keeper) {
+	lastObservedEventNonce := k.GetLastObservedEventNonce(ctx)
+	k.IterateEthereumEventVoteRecords(ctx, func(key []byte, eventVoteRecord *types.EthereumEventVoteRecord) bool {
+		event, err := types.UnpackEvent(eventVoteRecord.Event)
+		if err != nil {
+			panic(err)
+		}
+		eventNonce := event.GetEventNonce()
+		if eventNonce <= lastObservedEventNonce {
```

Comment:
Shouldn't this be `<` rather than `<=`?

Reply:
One thing I have noticed while helping validators diagnose issues with their orchestrators is that it's valuable to have evidence they have or have not been submitting things correctly. If we prune everything every time, it's very hard to query for this sort of information. I think leaving some small bit of history before pruning might be worthwhile.